### DISCUSSION TITLE
Android build issues

### DISCUSSION THREAD

**Discussion by dtrebilco** [StereoKit/StereoKit#203] [Ideas]
This is the steps I took to get this compiling on Android (for Quest2) in case anyone else has the same issues.
Also if any of this is useful for a future FAQ

- Just trying to build by default resulted in not finding xmake - so install xmake 
- xmake was not finding the Android SDK even though it was installed with visual studio. Seems that ANDROID_SDK_ROOT environment variable is not set when installing via Visual studio
- Lots of weird errors about unable to find "Files" when building. Visual studio had put the Android SDK in C:\Program Files (x86) and xmake or something it generated does not like the spaces. Moved the Android SDK to a path without spaces and re-targeted ANDROID_SDK_ROOT 
- Success - I now call adb install on the apk to copy it to the Quest2

**Comment by maluoi** [comment]
Thanks for the notes on your experience, this is good to know! I've had a little too much trouble with the Visual Studio version of the Android SDK, so I've been compiling the NuGet binaries with one provided by Google instead.

That bit about spaces is ah... disappointingly hilarious. It's been a while since I went through that, and I wonder if that's part of what stumped me originally 😬

**Comment by waruqi** [comment]
> xmake was not finding the Android SDK even though it was installed with visual studio. Seems that ANDROID_SDK_ROOT environment variable is not set when installing via Visual studio

xmake need not Android SDK, it's optional and only for depolying Qt app. It need only Android NDK.

```console
$ xmake f -p android --ndk=/Users/ruki/files/android-ndk-r21b
$ xmake
```

**Reply by dtrebilco** [reply]
Not an expert on this at all, but on Windows it put the NDK in a sub folder of the SDK and seems to use the ANDROID_SDK_ROOT environment var to find it.

**Reply by waruqi** [reply]
You can directly use `xmake f --ndk=` or `xmake g --ndk=` or ANDROID_NDK_ROOT to directly configure the ndk path.
ANDROID_SDK_ROOT is not necessary, only if the user does not specify the ndk path, xmake will try to detect other possible ndk paths, such as: ANDROID_SDK_ROOT/ndk-bundle
